File: recursion/max.py
def max(xs):
    if len(xs) == 1:
        return xs[0]
    else:
        sub_max = max(xs[1:])
        return xs[0] if xs[0] > sub_max else sub_max

print max([1]) # => 1
print max([1,2,10,3,4]) # => 10

File: recursion/max.js
function max(arr) {
  if (arr.length === 1) return arr[0];
  const [head, ...tail] = arr;
  const subMax = max(tail);
  return head > subMax ? head : subMax;
}

console.log(max([1])); // => 1
console.log(max([1, 2, 10, 3, 4])); // => 10
